### DISCUSSION TITLE
Fast way to merge multiple spheres

### DISCUSSION THREAD

**Discussion by duerrsimon** [pyvista/pyvista#1735] [Q&A]
I want construct a PolyData object from a few points (which can have different radii). 
The naive approach is a bit slow even for a relatively small number of points.  Execution time = 7.6s 

My Code 
```
import numpy as np
import pyvista as pv
points = np.array([[ 8.03 ,  0.595,  6.187],
       [ 5.916,  0.412,  6.479],
       [ 6.665,  1.744,  7.227],
       [ 6.458, -0.045,  6.066],
       [ 6.617,  0.392,  7.925],
       [ 6.478,  0.493,  5.789],
       [ 6.858,  1.304,  9.94 ],
       [ 7.068,  3.034, 11.187],
       [ 6.876,  2.689,  9.311],
       [ 7.062,  2.484, 11.436],
       [ 8.531,  2.534, 10.661],
       [ 6.403,  2.546, 10.901],
       [10.241,  3.528,  8.649],
       [10.431,  4.226,  6.632],
       [10.055,  2.668,  6.574],
       [10.504,  4.568,  7.457]])
sizes = np.random.uniform(2,3, size=(points.shape[0]))
spheres =  []
for i,p in enumerate(points):
    spheres.append(pv.Sphere(center=p,radius=sizes[i]))
result = spheres[0].boolean_union(spheres[1])
for s in spheres[2:]:
    result = result.decimate(0.7, volume_preservation=True) # this approximately halfs the duration
    result = result.boolean_union(s)
```

I assume this is because the outpout has many cells/points (decimating the result at each iteration makes it a bit faster). Any suggestions on how to make this faster? 
Perhaps an approach triangulating a point cloud is faster but here I am not sure on how to use different radii for the different spheres. Thanks!
<tr style="border-spacing: 0px; border-collapse: collapse; border: none !important; vertical-align: middle; text-align: right; color: rgb(212, 212, 212); font-family: system-ui, Ubuntu, &quot;Droid Sans&quot;, sans-serif; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><th style="border-spacing: 0px; border-collapse: collapse; border: none !important; vertical-align: middle; text-align: right; padding: 4px 8px; font-weight: normal;">Header</th><th style="border-spacing: 0px; border-collapse: collapse; border: none !important; vertical-align: middle; text-align: right; padding: 4px 8px; font-weight: normal;">Data Arrays</th></tr><tr style="border-spacing: 0px; border-collapse: collapse; border: none !important; vertical-align: middle; text-align: right; background-color: rgba(130, 130, 130, 0.08); color: rgb(212, 212, 212); font-family: system-ui, Ubuntu, &quot;Droid Sans&quot;, sans-serif; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><td style="border-spacing: 0px; border-collapse: collapse; border: none !important; padding: 4px 8px;">

PolyData | Information
-- | --
N Cells | 12014
N Points | 6266
X Bounds | 3.919e+00, 1.250e+01
Y Bounds | -2.031e+00, 6.554e+00
Z Bounds | 3.789e+00, 1.344e+01
N Arrays | 7

</td></tr>

**Comment by banesullivan** [comment]
Do you simply want to merge the meshes into one? You could go a different route with this and `combine` a `MultiBlock` dataset into an `UnstructuredGrid`, extract the surface topology, and then run a `clean` filter to merge duplicate points:

```py
merged = pv.MultiBlock(spheres).combine().extract_surface().clean()
```
`53.5 ms ± 1.37 ms per loop (mean ± std. dev. of 7 runs, 10 loops each)`

<img width="503" alt="Screen Shot 2021-11-26 at 12 49 47 AM" src="https://user-images.githubusercontent.com/22067021/143545588-590a3aae-4361-435d-8f39-fef61f6cdcdf.png">

**Reply by duerrsimon** [reply]
Thanks! This works just fine for getting the mesh. However, when I want to compute the intersection of two such merged meshes it no longer works: An example would be the following one where I am interested in getting the volume shared between white and red.  
![image](https://user-images.githubusercontent.com/16979577/144051130-fdd81867-6dc6-41dd-bbfd-dfc885dbcc58.png)
